addons/project/models/project_task.py

```diff
@@ -931,6 +931,9 @@ def create(self, vals_list):
                         default_project_id=project_id
                     ).default_get(['stage_id']).get('stage_id')
                 vals["stage_id"] = default_stage[project_id]
+                new_stage = self.env['project.task.type'].browse(vals['stage_id'])
+                if project_id not in new_stage.project_ids.ids:
+                    new_stage.write({'project_ids': [Command.link(project_id)]})
```

Comment:
Alternatively, I think we could do something like this when we create a project in project_id many2one field in quick create of the kanban. 🤔 

```diff
diff --git a/addons/project/models/project_task.py b/addons/project/models/project_task.py
index 7e2bbd67ed75..b0e547fdd937 100644
--- a/addons/project/models/project_task.py
+++ b/addons/project/models/project_task.py
@@ -905,9 +905,6 @@ class Task(models.Model):
                         default_project_id=project_id
                     ).default_get(['stage_id']).get('stage_id')
                 vals["stage_id"] = default_stage[project_id]
-                new_stage = self.env['project.task.type'].browse(vals['stage_id'])
-                if project_id not in new_stage.project_ids.ids:
-                    new_stage.write({'project_ids': [Command.link(project_id)]})
             # user_ids change: update date_assign
             if vals.get('user_ids'):
                 vals['date_assign'] = fields.Datetime.now()
diff --git a/addons/project/views/project_task_views.xml b/addons/project/views/project_task_views.xml
index 8d1348995848..5e46db7671fe 100644
--- a/addons/project/views/project_task_views.xml
+++ b/addons/project/views/project_task_views.xml
@@ -495,7 +495,7 @@
                     <group>
                         <field name="display_name" string= "Task Title" placeholder="e.g. Send Invitations"/>
                         <field name="project_id" widget="project_private_task" invisible="context.get('default_project_id', False)" placeholder="Private" class="o_project_task_project_field"
-                        domain="[('type_ids', 'in', context['default_stage_id'])] if context.get('default_stage_id') else []"/>
+                        domain="[('type_ids', 'in', context['default_stage_id'])] if context.get('default_stage_id') else []" context="{'default_type_ids': [(4, context.get('default_stage_id', False))]}"/>
                         <field name="user_ids" options="{'no_open': True, 'no_quick_create': True}" domain="[('share', '=', False), ('active', '=', True)]"
                             widget="many2many_avatar_user"/>
                         <field name="company_id" invisible="1"/>
```

Reply:
it works :-)